src/Core/Listener/EmailConfirmationMailer.php

```diff
@@ -11,20 +11,22 @@
 
 namespace Flarum\Core\Listener;
 
-use Flarum\Core;
 use Flarum\Core\EmailToken;
+use Flarum\Core\Jobs\MailJob;
+use Flarum\Core\Support\DispatchJobsTrait;
 use Flarum\Core\User;
 use Flarum\Event\UserEmailChangeWasRequested;
 use Flarum\Event\UserWasRegistered;
 use Flarum\Forum\UrlGenerator;
 use Flarum\Settings\SettingsRepositoryInterface;
+use Illuminate\Contracts\Bus\Dispatcher as JobDispatcher;
 use Illuminate\Contracts\Events\Dispatcher;
 use Illuminate\Contracts\Mail\Mailer;
-use Illuminate\Mail\Message;
 use Symfony\Component\Translation\TranslatorInterface;
 
 class EmailConfirmationMailer
 {
+    use DispatchJobsTrait;
```

Comment:
You might want to create an issue then to remove the need of a DispatchesEventTrait as well, it does exactly the same useless shortcut :) It might make more sense if the trait would load the bound Dispatcher too from my perspective.